invisible_cities/io/mcinfo_io.py

```diff
@@ -367,31 +581,43 @@ def load_mcsensor_response_df(file_name : str,
                Name of the detector database to be accessed
     run_no   : int
                Run number for database access
+    sns_name : str or list[str]
+               Sensor name of the data to be returned
+               Default: return for every sensor
     """
-    pmt_ids = DB.DataPMT(db_file, run_no).SensorID
-
-    pmt_bin, sipm_bin = get_sensor_binning(file_name)
+    if is_oldformat_file(file_name):
+        sns_resp = load_mcsensors_dfold(file_name)
+        if return_raw:
+            return sns_resp
+        assert(db_file), "Database name required for this file"
+        assert( run_no), "Run number required for database access"
+        return convert_timebin_to_time(sns_resp                          ,
+                                       get_sensor_binning     (file_name),
+                                       load_mcsensor_positions(file_name,
+                                                               db_file  ,
+                                                               run_no   ))
+    else:
+        sns_resp = load_dst(file_name, 'MC', 'sns_response')
+        if return_raw:
+            return sns_resp
+        return convert_timebin_to_time(sns_resp                          ,
+                                       get_sensor_binning     (file_name),
```

Comment:
I assume we can not have 'sns_response' without 'sns_position' table? But probably we do need a protection for that?

Reply:
Exactly, 'sns_response' is only filled with the positions of the sensors that have detected some photoelectron, at the moment. It's not a bad idea to add a protection, though, because we may want to save that table with all the sensor positions, in the future.

Reply:
In any case, all the tables are present in the file, always. They may be empty, though.

Reply:
Whether we have all the positions saved or just the ones that are hit doesn't affect this function since we only use the names of the detectors, not the positions etc.

In case of an empty sns_positions table there might be a fail path but I'm not 100% sure what it would be. I'll have a look to see if I can understand it a bit better